jwql/instrument_monitors/nirspec_monitors/ta_monitors/msata_monitor.py

```diff
@@ -0,0 +1,861 @@
+                          'FITXSIGM': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'lsv2sigma'},
+                          'FITYSIGM': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'lsv3sigma'},
+                          'ITERATNS': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'lsiterations'},
+                          'GUIDERID': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'guidestarid'},
+                          'IDEAL_X': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'guidestarx'},
+                          'IDEAL_Y': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'guidestary'},
+                          'IDL_ROLL': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'guidestarroll'},
+                          'SAM_X': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'samx'},
+                          'SAM_Y': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'samy'},
+                          'SAM_ROLL': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'samroll'},
+                          'box_peak_value': {'loc': 'ta_table', 'alt_key': None, 'name': 'box_peak_value'},
+                          'reference_star_mag': {'loc': 'ta_table', 'alt_key': None, 'name': 'reference_star_mag'},
+                          'convergence_status': {'loc': 'ta_table', 'alt_key': None, 'name': 'convergence_status'},
+                          'reference_star_number': {'loc': 'ta_table', 'alt_key': None, 'name': 'reference_star_number'},
+                          'lsf_removed_status': {'loc': 'ta_table', 'alt_key': None, 'name': 'lsf_removed_status'},
+                          'lsf_removed_reason': {'loc': 'ta_table', 'alt_key': None, 'name': 'lsf_removed_reason'},
+                          'lsf_removed_x': {'loc': 'ta_table', 'alt_key': None, 'name': 'lsf_removed_x'},
+                          'lsf_removed_y': {'loc': 'ta_table', 'alt_key': None, 'name': 'lsf_removed_y'},
+                          'planned_v2': {'loc': 'ta_table', 'alt_key': None, 'name': 'planned_v2'},
+                          'planned_v3': {'loc': 'ta_table', 'alt_key': None, 'name': 'planned_v3'},
+                          'FITTARGS': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'stars_in_fit'}}
+
+        # fill out the dictionary to create the dataframe
+        msata_dict = {}
+        for fits_file in new_filenames:
+            msata_info = self.get_tainfo_from_fits(fits_file)
+            if msata_info is None:
+                continue
+            main_hdr, ta_hdr, ta_table = msata_info
+            for key, key_dict in keywds2extract.items():
+                key_name = key_dict['name']
+                if key_name not in msata_dict:
+                    msata_dict[key_name] = []
+                ext = main_hdr
+                if key_dict['loc'] == 'ta_hdr':
+                    ext = ta_hdr
+                if key_dict['loc'] == 'ta_table':
+                    ext = ta_table
+                try:
+                    val = ext[key]
+                except:
```

Comment:
@penaguerrero do you know what exception you are trying to capture here? Is it a `KeyError`?